lib/hdcontroller.js

```diff
@@ -267,7 +270,16 @@ class HDProxydClient {
         assert.strictEqual(typeof key, 'string');
         const log = this.createLogger(reqUids);
         this._failover('DELETE', null, 0, key, 0, log, (err, res) => {
-            callback(err);
+            if (res) {
```

Comment:
On the commit message: I would explain *why* we need to drain the stream.

e.g.:

> Properly "consume" (aka drain) the IncomingMessage stream, otherwise the connections cannot be reused

Reply:
done